src/IW/Effects/Cabal.hs

```diff
@@ -0,0 +1,61 @@
+{- | This module contains the class definitiion of @MonadCabal@ and
+an instance of @MonadCabald@ for the @App@ monad. Instances of
+@MonadCabal@ have a @getCabalCategories@ action that returns @[Category]@
+given a @RepoOwner@ and @RepoName@.
+-}
+
+module IW.Effects.Cabal
+       ( getCabalCategories
+
+       -- * Internals
+       , getCabalCategoriesImpl
+       , repoCabalUrl
+       ) where
+
+import Data.Text (splitOn, strip)
+import Distribution.PackageDescription
+import Distribution.PackageDescription.Parsec (parseGenericPackageDescriptionMaybe)
+
+import IW.App (App)
+import IW.Core.Repo (RepoOwner (..), RepoName (..), Category (..))
+import IW.Core.Url (Url (..))
+import IW.Effects.Download (MonadDownload (..))
+
+
+-- | Describes a monad that returns @[Category]@ given a @RepoOwner@ and @RepoName@.
+class Monad m => MonadCabal m where
+    getCabalCategories :: RepoOwner -> RepoName -> m [Category]
+
+instance MonadCabal App where
+    getCabalCategories = getCabalCategoriesImpl
+
+getCabalCategoriesImpl :: MonadDownload m => RepoOwner -> RepoName -> m [Category]
+getCabalCategoriesImpl repoOwner repoName = do
+    cabalFile <- downloadFile $ repoCabalUrl repoOwner repoName
+    pure $ case parseGenericPackageDescriptionMaybe cabalFile of
```

Comment:
Let's add debugging log message that we are trying to parse *owner/repo/name.cabal* file now.

Also I assume that there is some logging about download, so we don't need to add logs here as well :slightly_smiling_face:

Reply:
@vrom911 Okay good I was wondering if we needed logging for both, but logging for `MonadDownload` should be enough, right?